snippets/toTitleCase.md

```diff
@@ -0,0 +1,19 @@
+### toTitleCase
+
+Converts a string to title case.
+
+Break the string into words, using a regexp, and combine them capitalizing the first letter of each word and adding a whitespace between them.
+
+```js
+const toTitleCase = str => str
+  .match(/[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g)
+  .map(x => x.charAt(0).toUpperCase() + x.slice(1).toLowerCase())
```

Comment:
Causes problems with things like 'mcDolans' becoming 'Mcdolans'.

Ideally since intermediate capitals should be preserved in title case, each word should just have the first letter capitalized

Reply:
```suggestion
  .map(x => x.charAt(0).toUpperCase() + x.slice(1))
```

Reply:
Ha, very good point! Thanks!